actionpack/lib/action_controller/metal/conditional_get.rb

```diff
@@ -157,8 +186,8 @@ def fresh_when(record_or_options, additional_options = {})
     #     super if stale? @article, template: 'widgets/show'
     #   end
     #
-    def stale?(record_or_options, additional_options = {})
-      fresh_when(record_or_options, additional_options)
+    def stale?(record_or_collection_or_options, additional_options = {})
```

Comment:
`record_or_collection_or_options` this is getting out of handy =) lol.
I would say this is a good opportunity to change the options passed to conditional_get to use kwargs! As far as I can tell we always validate by `:etag, :last_modified, :public, :template` , so those should be the args!

I also would say that refactor should happen first, and then we come back to evaluate this PR.